dao/src/main/java/org/thingsboard/server/dao/timeseries/BaseTimeseriesService.java

```diff
@@ -295,7 +295,12 @@ private void validate(ReadTsKvQuery query) {
             throw new IncorrectParameterException("Incorrect ReadTsKvQuery. Aggregation can't be empty");
         }
         if (!Aggregation.NONE.equals(query.getAggregation())) {
-            long step = Math.max(query.getInterval(), 1000);
+            long interval = query.getInterval();
+            if (interval < 1) {
+                throw new IncorrectParameterException("Invalid TsKvQuery: 'interval' must be greater than 0, but got " + interval +
+                        ". Please check your query parameters and ensure 'endTs' is greater than 'startTs' or increase 'interval'.");
```

Comment:
What endTs and startTs have to do with it?

Reply:
![image](https://github.com/user-attachments/assets/14a8f199-7905-4618-ba5c-4165211a879e)